postgis/postgis.sql.in

```diff
@@ -5970,7 +5970,44 @@ CREATE OR REPLACE FUNCTION ST_InterpolatePoint(Line geometry, Point geometry)
 	LANGUAGE 'c' IMMUTABLE STRICT PARALLEL SAFE
 	_COST_MEDIUM;
 
--- moved to separate file cause its invovled
+---------------------------------------------------------------
+-- Grid / Hexagon coverage functions
+---
+
+-- Availability: 3.0.0
+CREATE OR REPLACE FUNCTION ST_Hexagon(size float8, cell_i integer, cell_j integer, origin geometry DEFAULT 'POINT(0 0)')
+	RETURNS geometry
+	AS 'MODULE_PATHNAME', 'ST_Hexagon'
+	LANGUAGE 'c' IMMUTABLE STRICT
+	PARALLEL SAFE
+	_COST_MEDIUM;
+
+-- Availability: 3.0.0
+CREATE OR REPLACE FUNCTION ST_Square(size float8, cell_i integer, cell_j integer, origin geometry DEFAULT 'POINT(0 0)')
+	RETURNS geometry
+	AS 'MODULE_PATHNAME', 'ST_Square'
+	LANGUAGE 'c' IMMUTABLE STRICT
+	PARALLEL SAFE
+	_COST_MEDIUM;
+
+-- Availability: 3.0.0
+CREATE OR REPLACE FUNCTION ST_HexagonGrid(size float8, bounds geometry, OUT geom geometry, OUT i integer, OUT j integer)
```

Comment:
Are the OUT parameters in ST_HexagonGrid and ST_SquareGrid needed?

Reply:
Yes, that's my new trick to avoid having to declare "utility relations" just to support tuple-valued outputs. Works like a hot damn: declare output as anonymous record but use OUT parameters to provide name and types for the record members. If we want, we could excise all these old friends like `geometry_dump` `valid_detail` and such.